driver-core/src/main/com/mongodb/internal/dns/DefaultDnsResolver.java

```diff
@@ -64,13 +64,19 @@ public DefaultDnsResolver(@Nullable final DnsClient dnsClient) {
 
       The priority and weight are ignored, and we just concatenate the host (after removing the ending '.') and port with a
       ':' in between, as expected by ServerAddress.
-
-      It's required that the srvHost has at least three parts (e.g. foo.bar.baz) and that all of the resolved hosts have a parent
-      domain equal to the domain of the srvHost.
```

Comment:
Optional since this is internal, but consider replacing this with the revised conditions.